src/views/SearchOverlayView/SearchOverlayView.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { useEffect } from 'react'
+import { useRouterQuery } from '@/hooks'
+import RecentSearches from './RecentSearches'
+import SearchResults from './SearchResults'
+import styled from '@emotion/styled'
+import { colors, zIndex } from '@/shared/theme'
+import { TOP_NAVBAR_HEIGHT } from '@/components/TopNavbar'
+import { QUERY_PARAMS } from '@/config/routes'
+
+const SearchOverlayView: React.FC = () => {
+  const searchQuery = useRouterQuery(QUERY_PARAMS.SEARCH)
+
+  // TODO we should probably handle that in one place for all the overlays
+  // prevent body scroll
+  useEffect(() => {
+    document.body.style.setProperty('overflow', 'hidden')
```

Comment:
![](https://im6.ezgif.com/tmp/ezgif-6-e59e39e9e8e5.gif)
Not sure how it looks on another OS but It introduced the so-called [jumping scrollbar](https://aykevl.nl/2014/09/fix-jumping-scrollbar). In my opinion, it's not looking good. Layout shifts a bit when the overlay is showing up. The easiest solution is to just keep the scrollbar always visible, I guess.

Reply:
Damn, I was afraid this would happen. We need some fix for this since we will be introducing more overlays soon. Let me look into possible solutions

Reply:
Another simple solution would be adding `overflow-y: overlay;` in body.  Firefox doesn't support that though.

Reply:
Firefox marks it as deprecated and it's not standardized so that means we cannot really use it :(